File: builder/kotlin/LessonDirector.kt
package Builder

// Builder arayüzünü kullanarak nesne örneklemesi yapar.
// UML diyagramındaki Director'a denk gelmektedir.
class LessonDirector(private val lessonBuilder: LessonBuilder) {

    fun make() {
        lessonBuilder.getLesson()
        lessonBuilder.applyDiscount()
        lessonBuilder.addLessonNote()
    }
}

File: builder/php/LessonDirector.php
<?php

namespace Builder;

use Builder\LessonBuilder;

// Builder arayüzünü kullanarak nesne örneklemesi yapar.
// UML diyagramındaki Director'a denk gelmektedir.
class LessonDirector
{
    private LessonBuilder $lessonBuilder;

    public function __construct(LessonBuilder $lessonBuilder)
    {
        $this->lessonBuilder = $lessonBuilder;
    }

    public function make()
    {
        $this->lessonBuilder->getLesson();
        $this->lessonBuilder->applyDiscount();
        $this->lessonBuilder->addLessonNote();
    }
}
